src/docfx/moniker/ExpressionCreator.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Docs.Build
+{
+    internal class ExpressionCreator
+    {
+        private static readonly Regex OrSymbolRegex = new Regex(@"^\s*(?<or>\|\|)", RegexOptions.Compiled);
```

Comment:
can we parse these expressions without regex?

Reply:
using string spplit with `||` or comparators?

---
In reply to: [228094545](https://github.com/dotnet/docfx/pull/3529#discussion_r228094545) [](ancestors = 228094545)